pwiz_tools/Skyline/Model/Results/ChromatogramCache.cs

```diff
@@ -1230,6 +1231,91 @@ public IEnumerable<ChromKeyIndices> GetChromKeys(MsDataFileUri msDataFilePath)
             }
         }
 
+        private ChromatogramCache UpdateOptimizationSteps(SrmDocument doc)

```

Comment:
I think this function should only process MsDataFileUri's which belong to a ChromatogramSet with a non-null OptimizationFunction.